src/Enumerable.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Spatie\Enum;
+
+interface Enumerable
+{
+    /**
+     * Create a valued instance of the Enum by it's value or index.
+     *
+     * @param string|int $value
+     *
+     * @return \Spatie\Enum\Enumerable
+     */
+    public static function make($value): Enumerable;
+
+    /**
+     * Check if the current instance and the given value are equal.
+     *
+     * @param string|\Spatie\Enum\Enumerable $value
+     *
+     * @return bool
+     */
+    public function isEqual($value): bool;
+
+    /**
+     * Check if the current instance is equal with one of the given values.
+     *
+     * @param string[]|\Spatie\Enum\Enumerable[] $values
+     *
+     * @return bool
+     */
+    public function isAny(array $values): bool;
+
+    /**
+     * Check if the given index exists on this enum.
+     *
+     * @param int $index
+     *
+     * @return bool
+     */
+    public static function isIndex(int $index): bool;
```

Comment:
I find `isValidIndex` and `isValidValue` to be more clear as to what the methods actually do. What do you think?

Reply:
https://github.com/spatie/enum/pull/18/commits/8d8ccab2432a088a53d735bb6d7401343d6f0146